extension.ts

```diff
@@ -0,0 +1,303 @@
+              FUNCTION_LAYERS,
+              moduleName,
+              __routeFileName
+            );
+
+            await Promise.all(
+              Object.keys(_rOptions).map(async method => {
+                const _routekey = `${method} ${route}`;
+                _tRoutes[_routekey] = {};
+
+                await mkdir(_schemaDir, { recursive: true });
+                const _mOptions = _rOptions[method] as KeyOptions;
+
+                await Promise.all(
+                  Object.keys(_mOptions).map(async key => {
+                    const _options = _mOptions[key];
+                    _tRoutes[_routekey][key] = {};
+                    if (_options.schema) {
+                      const _sFileName = encodeFileSystem(_routekey, key);
+                      const _sFilePath = path.join(_schemaDir, _sFileName);
+                      await writeCompiledSchema(
+                        _sFilePath,
+                        _options.schema as JSONSchema7
+                      );
+                      _tRoutes[_routekey][key].schema = true;
+                    }
+
+                    if (_options.parser) {
+                      _tRoutes[_routekey][key].parser = _options.parser;
+                    }
+                  })
+                );
+              })
+            );
+
+            const _routesOutputPath = path.join(_schemaDir, ROUTE_FILE_NAME);
+
+            await writeFile(
+              _routesOutputPath,
+              "exports.routes = " + JSON.stringify(_tRoutes)
+            );
+
+            /**
+             * TODO: generate layer name accroding to somod
+             */
+            const layerResourceName =
+              "myLayer" + new Date().getUTCMilliseconds();
+
+            const layer = {
+              Type: AWS_SERVERLESS_LAYER_VERSION,
+              Properties: {
+                ContentUri: _schemaDir
+              }
+            };
+
+            let layers = (func["Properties"] as JSONObjectType)[
+              LAYERS
+            ] as Record<string, string>[];
+
+            if (!layers) {
+              layers = [];
+            }
+
+            console.log("layers");
+            console.log(layers);
+
+            layers.push({ Ref: layerResourceName });
+
+            console.log("layers updated");
+            console.log(layers);
+
+            (template["Resources"] as JSONObjectType)[layerResourceName] =
+              layer;
+          })
+        );
+      }
+    })
+  );
+  updateYamlFileStore(templateFile, template);
+  saveYamlFileStore(templateFile);
+  console.log("prepare Ended");
+};
+
+const writeCompiledSchema = async (fileName: string, schema: JSONSchema7) => {
+  const standaloneValidator = await getCompiledValidator(schema);
+
+  await writeFile(fileName, standaloneValidator);
+};
+
+const getRouteFiles = async ({
+  module
+}: {
+  module: Module;
+}): Promise<string[]> => {
+  const functionsDir = path.join(module.packageLocation, SERVERLESS, FUNCTIONS);
```

Comment:
if `module.root == true` then the files are available at `serverless/functions/**.http.yaml`
otherwise , the files are at `build/serverless/functions/**.http.json`

Reply:
fixed